lib/journald/reader.go

```diff
@@ -157,3 +164,20 @@ func (r *reader) getString(k string) (s string) {
 	s, _ = r.GetDataValue(k)
 	return
 }
+
+func sanitizeStreamName(name string) string {
+	name = strings.Replace(name, ":", "/", -1)
+	name = strings.Replace(name, "*", "/", -1)
+	max := len(name)
+	if max > 512 {
+		max = 512
+	}
+	return name[:max]
+}
+
+func arnToUUID(task string) string {
+	// ECS Task ARN looks like:
+	// arn:aws:ecs:<region>:<aws_account_id>:task/<UUID>
+	parts := strings.Split(task, "/")
```

Comment:
This function could be implemented more efficiently using `strings.IndexByte(task, '/')`.